From 82dddcd1462e1d46f9c727f4f2f6738f47c8de13 Mon Sep 17 00:00:00 2001
Message-ID: <82dddcd1462e1d46f9c727f4f2f6738f47c8de13.1710883391.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1710883390.git.sharlatanus@gmail.com>
References: <cover.1710883390.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 19 Mar 2024 21:16:04 +0000
Subject: [PATCH 08/10] gnu: Add cl-vgplot.

* gnu/packages/lisp-xyz.scm (cl-vgplot, ecl-vgplot, sbcl-vgplot): New
variables.

Change-Id: I0c5aa5f3a0bb5bf79f0750b957442533073e5e2c
---
 gnu/packages/lisp-xyz.scm | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index 6c59dd915e..81a0a164ae 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -7711,6 +7711,40 @@ (define-public ecl-verbose
 (define-public cl-verbose
   (sbcl-package->cl-source-package sbcl-verbose))
 
+(define-public sbcl-vgplot
+  (let ((commit "76329928a3709387c0779175af56ec49d16bcd35")
+        (revision "0"))
+    (package
+      (name "sbcl-vgplot")
+      (version (git-version "0.0.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/volkers/vgplot")
+               (commit commit)))
+         (file-name (git-file-name "vgplot" version))
+         (sha256
+          (base32 "1vc5fd787xa8831wjbmwrpg17f9isi5k8dmb85fsysz47plbvi1y"))))
+      (build-system asdf-build-system/sbcl)
+      (native-inputs
+       (list sbcl-lisp-unit))
+      (inputs
+       (list sbcl-cl-fad sbcl-cl-ppcre sbcl-ltk))
+      (home-page "https://github.com/volkers/vgplot")
+      (synopsis "Common lisp library interfacing to the gnuplot utility")
+      (description
+       "This package provides an interface to the @code{gnuplot} plotting
+utility.  The intention of the API is to resemble to some of the plot commands
+of octave or matlab.")
+      (license license:gpl3))))
+
+(define-public cl-vgplot
+  (sbcl-package->cl-source-package sbcl-vgplot))
+
+(define-public ecl-vgplot
+  (sbcl-package->ecl-package sbcl-vgplot))
+
 (define-public sbcl-find-port
   (let ((commit "811727f88d7f000623bf92fdb0e64678a7112a28")
         (revision "2"))
-- 
2.41.0

